src/scripts/components/main/globe/globe.tsx

```diff
@@ -1,395 +1,374 @@
-// keep a reference to the current basemap layer
-let basemapLayer: ImageryLayer | null = null;
+const EMPTY_FUNCTION = () => {};
 
-function getBasemapUrl(layerDetails: Layer | null) {
-  // set defaultBasemap when no layer is selected and defaultLayerBasemap if layer has no own basemap
-  if (!layerDetails) {
-    return isElectron()
-      ? config.basemapUrlsOffline[config.defaultBasemap]
-      : config.basemapUrls[config.defaultBasemap];
-  } else if (
-    !layerDetails.basemap ||
-    !config.basemapUrls[layerDetails.basemap]
-  ) {
-    return isElectron()
-      ? config.basemapUrlsOffline[config.defaultLayerBasemap]
-      : config.basemapUrls[config.defaultLayerBasemap];
-  }
+const Globe: FunctionComponent<Props> = props => {
+  const {
+    projectionState,
+    onMouseEnter,
+    onTouchStart,
+    layerDetails,
+    imageLayer,
+    markers
+  } = props;
 
-  return isElectron()
-    ? config.basemapUrlsOffline[layerDetails.basemap]
-    : config.basemapUrls[layerDetails.basemap];
-}
+  const [containerRef, globe] = useWebGlGlobe();
+  const initialTilesLoaded = useInitialBasemapTilesLoaded(globe);
 
-const Globe: FunctionComponent<Props> = ({
-  view,
-  projectionState,
-  imageLayer,
-  layerDetails,
-  spinning,
-  active,
-  flyTo,
-  markers = [],
-  backgroundColor,
-  onMouseEnter,
-  onTouchStart,
-  onChange,
-  onMoveEnd,
-  onMouseDown
-}) => {
-  const [viewer, setViewer] = useState<Viewer | null>(null);
-  const [firstTilesLoaded, setFirstTilesLoaded] = useState(false);
-  const ref = useRef<HTMLDivElement>(null);
-  const lastNowRef = useRef<number | null>(null);
-
-  // make latest "active" value always accessible in camera change handler
-  const isActiveRef = useRef<boolean>(active);
-  isActiveRef.current = active;
-
-  const spin = useCallback(() => {
-    if (!viewer) {
-      return;
-    }
-    const now = Date.now();
-    const spinRate = 0.08;
-    const delta = (now - (lastNowRef?.current ?? 0)) / 1000;
-    lastNowRef.current = now;
-    viewer.scene.camera.rotate(Cartesian3.UNIT_Z, spinRate * delta);
-  }, [viewer]);
-
-  // init cesium viewer
-  useEffect(() => {
-    if (!ref || !ref.current) {
-      return () => {};
-    }
+  useGlobeLayers(globe, layerDetails, imageLayer);
+  useGlobeMarkers(globe, markers);
 
-    // set correct scene mode
-    const sceneMode =
-      projectionState.projection === GlobeProjection.Sphere
-        ? SceneMode.SCENE3D
-        : SceneMode.SCENE2D;
-
-    // create default imagery provider
-    const defaultBasemapImageryProvider = new TileMapServiceImageryProvider({
-      url: getBasemapUrl(layerDetails),
-      fileExtension: 'png',
-      maximumLevel: 4
-    });
+  useProjectionSwitch(globe, projectionState.projection);
+  useMultiGlobeSynchronization(globe, props);
 
-    const options = {
-      ...cesiumOptions,
-      sceneMode,
-      imageryProvider: defaultBasemapImageryProvider
-    };
+  // fixme: right now @ubilabs/esa-webgl-globe doesn't support the
```

Comment:
I think these events are handled by react in line 88+89 so no need to implement this in the globe itself. These are just listeners on the containing div to decide which of the two globes is the active one if I remember correctly.